src/mscorlib/shared/System/String.Searching.cs

```diff
@@ -79,28 +82,77 @@ public unsafe int IndexOf(char value, int startIndex, int count)
 
             fixed (char* pChars = &_firstChar)
             {
-                char* pCh = pChars + startIndex;
+                char* pStartCh = pChars + startIndex;
+                char* pCh = pStartCh;
 
-                while (count >= 4)
+                int nLength = count;
+                bool useVectorization = false;
```

Comment:
Is there a good reason why this is using this bool variable? https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanHelpers.byte.cs#L96 does not use it.

Reply:
My reasoning was so it wouldn't have to do the index calculation and checks when we aren't using the vectorized approach (ex. when the string is too short).

https://github.com/dotnet/coreclr/pull/16392/files#diff-f7e6389a6519754f04d20215ff42efcdR126

The SpanHelpers approach uses a `ref byte searchSpace` and keeps track of the index.  Here I kept the same `char*` pointer approach.  So to get the index, I either need to calculate it, or also keep track of the index as we go.

Doing this saved ~.5ns in the `IndexOfShort_Match` case, and ~1ns in the `IndexOfShort_Miss` case on my machine.

Reply:
There are two long-lived variables: pStartCh and useVectorization. They will burn two registers or two stack spill slots. I think it maybe be more efficient if you replace them by just one `pEndCh`.

Reply:
Do you need both `pCh` and `index`?